target/linux/ramips/patches-6.6/845-pwm-add-mediatek-support.patch

```diff
@@ -172,49 +172,26 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
 +
 +static const struct pwm_ops mtk_pwm_ops = {
 +	.apply = mtk_pwm_apply,
-+	.owner = THIS_MODULE,
 +};
 +
 +static int mtk_pwm_probe(struct platform_device *pdev)
 +{
 +	struct mtk_pwm_chip *pc;
-+	struct resource *r;
-+	int ret;
 +
 +	pc = devm_kzalloc(&pdev->dev, sizeof(*pc), GFP_KERNEL);
 +	if (!pc)
 +		return -ENOMEM;
 +
-+	r = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-+	pc->mmio_base = devm_ioremap_resource(&pdev->dev, r);
++	pc->mmio_base = devm_platform_ioremap_resource(pdev, 0);
 +	if (IS_ERR(pc->mmio_base))
 +		return PTR_ERR(pc->mmio_base);
 +
-+	platform_set_drvdata(pdev, pc);
-+
 +	pc->chip.dev = &pdev->dev;
 +	pc->chip.ops = &mtk_pwm_ops;
 +	pc->chip.base = -1;
 +	pc->chip.npwm = NUM_PWM;
 +
-+	ret = pwmchip_add(&pc->chip);
-+	if (ret < 0)
-+		dev_err(&pdev->dev, "pwmchip_add() failed: %d\n", ret);
-+
-+	return ret;
-+}
-+
-+static int mtk_pwm_remove(struct platform_device *pdev)
-+{
-+	struct mtk_pwm_chip *pc = platform_get_drvdata(pdev);
-+	int i;
-+
-+	for (i = 0; i < NUM_PWM; i++)
-+		pwm_disable(&pc->chip.pwms[i]);
```

Comment:
Why are you removing this `pwm_disable` call?

Reply:
I would think devm handles this

Reply:
I haven't seen that in the code.

Reply:
Apparently to remove this, a switch to the async API is needed. I'm not sure on the details. Added back.